lib/kcerts/keys.go

```diff
@@ -8,22 +8,17 @@ import (
 	"crypto/x509"
 	"encoding/pem"
 	"golang.org/x/crypto/ssh"
-	"strings"
 )
 
-type PrivateKey struct {
-	Key PrivateKeyProvider
-}
-
-func (e PrivateKey) Signer() (ssh.Signer, error) {
-	res, err := NewSSHSigner(e.Key.Signer(), e.Key.SigningAlgo())
+func NewSigner(key PrivateKey) (ssh.Signer, error) {
+	res, err := NewSSHSigner(key.Signer(), key.SigningAlgo())
```

Comment:
nit: Make this a one liner?
```
return NewSSHSigner(...)
```

Also... the only thing this function does is call `.Signer()` and `.SigningAlgo()` for us? not sure it has much value, I would consider just removing it.

Reply:
I'm going to keep it in here to mimic the ssh.NewSignerFromSigner function. I'm keeping it in because I'm going to be getting more use out of the factory func once mTLS hits.